### PR TITLE
MRG, BUG: Raise error on normal for wrong src.kind

### LINKED ISSUE #6803: ENH: Prevent users from making some poor choices in volumes
From the [gitter](https://gitter.im/mne-tools/mne-python?at=5d8780f3c77f285fb1c2ede4). In a `VolVectorSourceEstimate` or a `VolSourceEstimate`, attempting to use the inherited `normal` probably should at throw an error, or at least a warning.  It might also be good for the various `apply_inverse` functions to do something similar in these spaces when `pick_ori='normal'`.  Looks like `_compute_beamformer` might already do this.

### PR DESCRIPTION
**Pull request by larsoner**
Closes #6803.

Didn't change any examples just tried to touch the ones that were likely to be affected.

### CODE REVIEW

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -113,6 +113,8 @@ Bug
 
 - Fix bug in :meth:`mne.SourceMorph.apply` where the default was errantly ``mri_space=False`` instead of ``mri_space=None`` (as documented), by `Eric Larson`_
 
+- Fix :meth:`mne.VolVectorSourceEstimate.normal` and :func:`mne.minimum_norm.apply_inverse` to only allow normal extrcation for surface and discrete source spaces by `Eric Larson`_
```
Comment (drammock) [inline doc/changes/latest.inc]: "extrcation" (typo). Should be "orientation"?

Comment (agramfort) [inline doc/changes/latest.inc]:
```suggestion
- Fix :meth:`mne.VolVectorSourceEstimate.normal` and :func:`mne.minimum_norm.apply_inverse` to only allow normal extraction for surface and discrete source spaces by `Eric Larson`_
```

**Comment by larsoner** [comment]
> I am not sure I would have allowed it for a discrete source space. Any reason why you think it's a good idea to accept this?

Yes because a discrete source space requires explicitly defining two things: positions and normals. Pretty much the only time I've seen it used is when for example you want to work with a single dipole or a set of dipoles and you know the position and orientation. Hence `discrete` I see as a sort of expert option. It wouldn't make sense to me to prevent this use case, even if there are others where people arbitrarily set the normals.

**Comment by agramfort** [comment]
thx @larsoner !

**Pull request merged**